account_invoice_report_grouped_by_picking/views/report_invoice.xml

```diff
@@ -0,0 +1,67 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+<template id="report_invoice_document" inherit_id="account.report_invoice_document" priority="99">
+    <xpath expr="//tbody[@class='invoice_tbody']" position="before">
+        <t t-set="subtotal" t-value="0.0"/>
+    </xpath>
+    <xpath expr="//tr[@t-foreach='o.invoice_line_ids']" position="attributes">
+        <attribute name="t-foreach">o.lines_grouped_by_picking()</attribute>
+        <attribute name="t-as">lines_group</attribute>
+    </xpath>
+    <xpath expr="//td/span[@t-field='l.name']/.." position="before">
+        <t t-set="l" t-value="lines_group['line']"/>
+        <t t-set="picking" t-value="lines_group['picking']"/>
+        <t t-if="picking != last_picking">
```

Comment:
This should work for all the pickings including last, so you need to change the logic and put it at the end instead at the beginning.